arena_test.go

```diff
@@ -0,0 +1,76 @@
+package bigqueue
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestNewArenaNoDir(t *testing.T) {
+	arena, err := newArena(fmt.Sprintf("%d/temp.dat", time.Now().UnixNano()), 100)
+	if arena != nil || err == nil || !strings.Contains(err.Error(), "no such file or directory") {
+		t.Errorf("unexpected return for newArena :: %v", err)
+	}
+}
+
+func TestNewArenaNoFile(t *testing.T) {
+	arenaSize := 100
+	fileName := path.Join(os.TempDir(), "temp.dat")
+	defer os.Remove(fileName)
+
+	arena, err := newArena(fileName, arenaSize)
+	if err != nil {
+		t.Errorf("error in creating new arena: %v", err)
+	}
+	defer arena.Unmap()
+
+	// ensure arena struct stores correct size
+	if arena.size != arenaSize {
+		t.Errorf("arena size do not match, expected: %v, actual: %v", arenaSize, arena.size)
+	}
+
+	// ensure underlined file is of correct size
+	info, err := os.Stat(fileName)
+	if err != nil {
+		t.Errorf("error in getting stats for file: %v", err)
+	}
+	if int(info.Size()) != arenaSize {
+		t.Errorf("arena file size do not match, expected: %v, actual: %v", arenaSize, info.Size())
+	}
+}
+
+func TestNewArenaLargerFile(t *testing.T) {
+	arenaSize := 100
+	fileName := path.Join(os.TempDir(), "temp.dat")
+	defer os.Remove(fileName)
```

Comment:
test should fail, if os.Remove returns error.

Reply:
Not really. This is only a cleanup step of the test